pkg/aws/client/client.go

```diff
@@ -304,44 +306,29 @@ func (c *Client) GetDHCPOptions(ctx context.Context, vpcID string) (map[string]s
 // DeleteObjectsWithPrefix deletes the s3 objects with the specific <prefix> from <bucket>. If it does not exist,
 // no error is returned.
 func (c *Client) DeleteObjectsWithPrefix(ctx context.Context, bucket, prefix string) error {
-	input := &s3.ListObjectsV2Input{
+	bucketVersioningStatus, err := c.S3.GetBucketVersioning(ctx, &s3.GetBucketVersioningInput{
 		Bucket: aws.String(bucket),
-		Prefix: aws.String(prefix),
-	}
-
-	paginator := s3.NewListObjectsV2Paginator(&c.S3, input)
-	for paginator.HasMorePages() {
-		objectIDs := make([]s3types.ObjectIdentifier, 0)
-		output, err := paginator.NextPage(ctx)
-		if err != nil {
-			return err
-		}
-		for _, object := range output.Contents {
-			identifier := s3types.ObjectIdentifier{Key: object.Key}
-			objectIDs = append(objectIDs, identifier)
-		}
-		if len(objectIDs) != 0 {
-			if _, err = c.S3.DeleteObjects(ctx, &s3.DeleteObjectsInput{
-				Bucket: aws.String(bucket),
-				Delete: &s3types.Delete{
-					Objects: objectIDs,
-					Quiet:   aws.Bool(true),
-				},
-			}); err != nil {
-				var nsk *s3types.NoSuchKey
-				if errors.As(err, &nsk) {
-					return nil
-				}
-				return err
+	})
+	if err != nil {
+		var apiErr smithy.APIError
+		if errors.As(err, &apiErr) {
+			if apiErr.ErrorCode() == "NoSuchBucket" {
+				// bucket doesn't exist, no action required
+				return nil
```

Comment:
I’m seeing the same NoSuchBucket check in multiple places. Could we pull this into a small helper function

Reply:
addressed you review comments.